de.metas.adempiere.adempiere/base/src/main/java/de/metas/banking/api/IBPBankAccountDAO.java

```diff
@@ -44,11 +44,8 @@
 	 * Retrieve all the bank accounts of the currency <code>currencyID</code> for the partner <code> partnerID</code>
 	 * In case the currencyID is not set (<=0) just retrieve all accounts of the bpartner
 	 * The bank accounts will be ordered by their IsDefault values, with true first.
-	 * 
-	 * @param ctx
-	 * @param partner
-	 * @param currency
-	 * @return
 	 */
 	List<I_C_BP_BankAccount> retrieveBankAccountsForPartnerAndCurrency(Properties ctx, int partnerID, int currencyID);
+
+	Optional<BankAccountId> retrieveBankAccountByBPartnerAndCurrencyAndIBAN(@NonNull BPartnerId bPartnerId, @NonNull CurrencyId currencyId, @NonNull String iban);
```

Comment:
good. different BPartners might have the same IBAN. Because, e.g. there might be in reality just one customer with one IBAN, but the metasfresh-user might want to map revenue onto two different C_BPartners, e.g. for book-keeping and reporting